lib/internal/modules/esm/translators.js

```diff
@@ -125,23 +133,92 @@ translators.set('module', async function moduleStrategy(url) {
 // Strategy for loading a node-style CommonJS module
 const isWindows = process.platform === 'win32';
 const winSepRegEx = /\//g;
-translators.set('commonjs', function commonjsStrategy(url, isMain) {
+translators.set('commonjs', async function commonjsStrategy(url, isMain) {
```

Comment:
Why does this function need to be async?

Reply:
All module strategies are promises. This was originally due to the Wasm path which is async, but strictly speaking all of these should really be async functions by default even if sync. They just get wrapped in Promise.resolve() anyway otherwise.